Warning Shown for Ignored Files

SublimeLinter-eslint is showing a generic linter warning on ignored files.

I'm ignoring this file intentionally so eslint doesn't bother (it's a data file in this case).

#### `SublimeLinter` console output

```
SublimeLinter: eslint: nested_arguments.js ['$REPO/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
SublimeLinter: eslint output:
$REPO/tests/fixtures/nested_arguments.js: line 0, col 0, Warning - File ignored because of a matching ignore pattern. Use "--no-ignore" to override.

1 problem 
```

#### `eslint` output from command line

```
kstone@Kevins-MacBook-Pro $REPO> ./node_modules/.bin/eslint tests/fixtures/nested_arguments.js 

$REPO/tests/fixtures/nested_arguments.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

✖ 1 problem (0 errors, 1 warning)
```

#### Additional info if needed (ST version, OS, configuration)

--- a/.flake8
+++ b/.flake8
@@ -1,0 +1,12 @@
+[flake8]
+max-line-length = 120
+
+# D100  Missing docstring in public module
+# D101  Missing docstring in public class
+# D102  Missing docstring in public method
+# D103  Missing docstring in public function
+# D104  Missing docstring in public package
+# D105  Missing docstring in magic method
+# D107  Missing docstring in __init__
+ignore = D100,D101,D102,D103,D105,D107
+

--- a/README.md
+++ b/README.md
@@ -39,11 +39,9 @@ Reopen your project next (or restart ST) to make sure local `eslint` will be use
 
 1. If you are using `zsh` and `oh-my-zsh`, do not load the `nvm` plugin for `oh-my-zsh`.
 
-Once `eslint` is installed, you must ensure it is in your system PATH so that SublimeLinter can find it. This may not be as straightforward as you think, so please read about [how linter executables are located][locating-executables] in the documentation.
-
 Once you have installed `eslint` you can proceed to install the SublimeLinter-eslint plugin if it is not yet installed.
 
-**Note:** This plugin requires `eslint` 1.0.0 or later.
+**Note:** This plugin requires `eslint` 2.0.0 or later.
 
 ### Plugin installation
 Please use [Package Control][pc] to install the linter plugin. This will ensure that the plugin will be updated when new versions are available. If you want to install from source so you can modify the source code, you probably know what you are doing so we won’t cover that here.
@@ -71,7 +69,7 @@ Use SublimeText console and SublimeLinter debug mode.
 Then open any JS file and run `Tools -> SublimeLinter -> Lint This View`. It must be an output in console after, something like that:
 
 ```
-SublimeLinter: eslint: 1.js ['/Projects/sample/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@'] 
+SublimeLinter: eslint: 1.js ['/Projects/sample/node_modules/.bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@']
 ```
 
 ##### I've got 'SublimeLinter: ERROR: eslint cannot locate 'eslint' in ST console when I try to use locally installed `eslint`.
@@ -86,78 +84,6 @@ npm install eslint
 ##### Plugin still does not work or there are errors in ST console.
 
 Update `eslint` instance, probably you use outdated version and SublimeLinter does not check it properly sometimes.
-
-##### There are no errors in console, but plugin does nothing.
-
-ESLint >2.0.0 does not enable any default rules and you should have config file for your code. Run in your console:
-```
-eslint --init # if eslint is global
-./node_modules/.bin/eslint --init # if eslint is installed locally
-```
-
-##### I want plugin to use my `.eslintignore` settings.
-
-~~It does it now.~~
-
-###### For ESLint <2.0.0
-
-Add to your SublimeLinter settings (global or per-project):
-
-```json
-{
-    "linters": {
-        "eslint": {
-            "args": [
-                "--stdin-filename", "__RELATIVE_TO_FOLDER__"
-            ]
-        }
-    }
-}
-```
-
-It can limit some rules and probably cause some bugs (i.e. files in symlinked folders can be skipped).
-
-###### For ESLint >=2.0.0 or if previous solution can't be applied
-
-Duplicate ESLint ignore settings to your `.sublimelinterrc` file. Use double stars for correct match.
-
-```json
-{
-    "linters": {
-        "eslint": {
-            "excludes": [
-                "**/node_modules/**",
-                "**/vendor/**"
-            ]
-        }
-    }
-}
-
-```
-
-###### For ESLint >=2.0.0
-
-If you use a SublimeText project and `.sublime-project` file is in project folder, set `chdir` to `${folder}` in your SublimeLinter settings. **Warning:** it can cause bugs if your project has more than one root folder.
-
-##### I want not to lint files if there is no `.eslintrc` file in project folder (for ESLint <1.0.0).
-
-Use `--reset` [ESLint](http://eslint.org/docs/user-guide/command-line-interface#reset) option, add it to your SublimeLinter global settings or project `.sublimelinterrc` file as below. Add `--no-reset` option to project `.sublimelinterrc` to overwrite it back.
-
-```json
-{
-    "linters": {
-        "eslint": {
-            "args": [
-                "--reset"
-            ]
-        }
-    }
-}
-```
-
-##### I want to use global `.eslintrc` config.
-
-Plugin uses the same [configuration hierarchy](http://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy) as `eslint` itself, so add `.eslintrc` to your home directory or ancestor directory of project.
 
 ##### I want to use custom rules, global `.eslintignore` file, etc.
 

--- a/linter.py
+++ b/linter.py
@@ -22,9 +22,7 @@ class ESLint(NodeLinter):
     syntax = ('javascript', 'html', 'json')
     npm_name = 'eslint'
     cmd = ('eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@')
-    version_args = '--version'
-    version_re = r'v(?P<version>\d+\.\d+\.\d+)'
-    version_requirement = '>= 1.0.0'
+
     regex = (
         r'^.+?: line (?P<line>\d+), col (?P<col>\d+), '
         r'(?:(?P<error>Error)|(?P<warning>Warning)) - '

--- a/linter.py
+++ b/linter.py
@@ -17,7 +17,6 @@ from SublimeLinter.lint import NodeLinter
 
 
 class ESLint(NodeLinter):
-
     """Provides an interface to the eslint executable."""
 
     syntax = ('javascript', 'html', 'javascriptnext', 'javascript (babel)',
@@ -51,7 +50,6 @@ class ESLint(NodeLinter):
 
         We override this method to handle parsing eslint crashes.
         """
-
         match = self.config_fail_regex.match(output)
         if match:
             return [(match, 0, None, "", "config", match.group(1), None)]
@@ -67,7 +65,6 @@ class ESLint(NodeLinter):
 
         We override this method to silent warning by .eslintignore settings.
         """
-
         v1message = 'File ignored because of your .eslintignore file. Use --no-ignore to override.'
         v2message = 'File ignored because of a matching ignore pattern. Use --no-ignore to override.'
 
@@ -79,7 +76,6 @@ class ESLint(NodeLinter):
 
     def communicate(self, cmd, code=None):
         """Run an external executable using stdin to pass code and return its output."""
-
         if '__RELATIVE_TO_FOLDER__' in cmd:
 
             relfilename = self.filename

--- a/linter.py
+++ b/linter.py
@@ -21,7 +21,7 @@ logger = logging.getLogger('SublimeLinter.plugin.eslint')
 class ESLint(NodeLinter):
     """Provides an interface to the eslint executable."""
 
-    syntax = ('javascript', 'html', 'json')
+    syntax = ('javascript', 'html')
     npm_name = 'eslint'
     cmd = ('eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@')
 

--- a/linter.py
+++ b/linter.py
@@ -19,8 +19,7 @@ from SublimeLinter.lint import NodeLinter
 class ESLint(NodeLinter):
     """Provides an interface to the eslint executable."""
 
-    syntax = ('javascript', 'html', 'javascriptnext', 'javascript (babel)',
-              'javascript (jsx)', 'json', 'jsx-real', 'vue component', 'vue')
+    syntax = ('javascript', 'html', 'json')
     npm_name = 'eslint'
     cmd = ('eslint', '--format', 'compact', '--stdin', '--stdin-filename', '@')
     version_args = '--version'

--- a/linter.py
+++ b/linter.py
@@ -10,8 +10,12 @@
 
 """This module exports the ESLint plugin class."""
 
+import logging
 import re
 from SublimeLinter.lint import NodeLinter
+
+
+logger = logging.getLogger('SublimeLinter.plugin.eslint')
 
 
 class ESLint(NodeLinter):
@@ -26,10 +30,6 @@ class ESLint(NodeLinter):
         r'(?:(?P<error>Error)|(?P<warning>Warning)) - '
         r'(?P<message>.+)'
     )
-    config_fail_regex = re.compile(
-        r'.*(ESLint couldn\'t find a configuration file)',
-        re.DOTALL
-    )
     crash_regex = re.compile(
         r'^(.*?)\r?\n\w*(Oops! Something went wrong!)',
         re.DOTALL
@@ -40,17 +40,14 @@ class ESLint(NodeLinter):
     }
 
     def find_errors(self, output):
-        """
-        Parse errors from linter's output.
+        """Parse errors from linter's output.
 
-        We override this method to handle parsing eslint crashes.
+        Log errors when eslint crashes or can't find its configuration.
         """
-        match = self.config_fail_regex.match(output)
-        if match:
-            return [(match, 0, None, "", "config", match.group(1), None)]
         match = self.crash_regex.match(output)
         if match:
-            return [(match, 0, None, "exception", "", match.group(2), None)]
+            logger.error(output)
+            return []
 
         return super().find_errors(output)
 

--- a/linter.py
+++ b/linter.py
@@ -10,8 +10,6 @@
 
 """This module exports the ESLint plugin class."""
 
-import sublime
-import os
 import re
 from SublimeLinter.lint import NodeLinter
 
@@ -70,25 +68,3 @@ class ESLint(NodeLinter):
             return match, None, None, None, None, '', None
 
         return match, line, col, error, warning, message, near
-
-    def communicate(self, cmd, code=None):
-        """Run an external executable using stdin to pass code and return its output."""
-        if '__RELATIVE_TO_FOLDER__' in cmd:
-
-            relfilename = self.filename
-            window = self.view.window()
-
-            # can't get active folder, it will work only if there is one folder in project
-            if int(sublime.version()) >= 3080 and len(window.folders()) < 2:
-
-                vars = window.extract_variables()
-
-                if 'folder' in vars:
-                    relfilename = os.path.relpath(self.filename, vars['folder'])
-
-            cmd[cmd.index('__RELATIVE_TO_FOLDER__')] = relfilename
-
-        elif not code:
-            cmd.append(self.filename)
-
-        return super().communicate(cmd, code)

--- a/linter.py
+++ b/linter.py
@@ -52,16 +52,12 @@ class ESLint(NodeLinter):
         return super().find_errors(output)
 
     def split_match(self, match):
-        """
-        Extract and return values from match.
+        """Extract and return values from match.
 
-        We override this method to silent warning by .eslintignore settings.
+        Return 'no match' for ignored files
         """
-        v1message = 'File ignored because of your .eslintignore file. Use --no-ignore to override.'
-        v2message = 'File ignored because of a matching ignore pattern. Use --no-ignore to override.'
-
         match, line, col, error, warning, message, near = super().split_match(match)
-        if message and (message == v1message or message == v2message):
+        if message and message.startswith('File ignored'):
             return match, None, None, None, None, '', None
 
         return match, line, col, error, warning, message, near

